docs/src/main/asciidoc/_admin-guide/management-tasks/Configuration_file_git_history.adoc

```diff
@@ -0,0 +1,138 @@
+[[Configuration_file_git_history]]
+= Git Configuration file history
+
+To enhance the initial <<Configuration_file_history],configuration file history>> we have now a native Git support to manage the configuration history. This feature goes a little farther than the initial configuration file history in that it also manages content and all the configuration files (such as properties). This feature only work for **standalone servers** using the default directory layout.
+
+As mentioned in <<Command_line_parameters,Command line parameters>> we support the usage of a remote Git repository to pull the configuration from or create or use a local Git repository.
+In fact if a `.git` directory exists under `jboss.server.base.dir` then using Git for managing configuration files will be automatically activated.
+Each modification of the content or the configuration will result in a new commit when the operation is successful and there are changes to commit. If there is an authenticated user then it will be stored as the author of the commit.
+Please note that this is a real Git repository so using a native Git client you can manipulate it.
+
+Now if we execute a management operation that modifies
+the model, for example adding a new system property using the CLI:
+
+[source, ruby]
+----
+[standalone@localhost:9990 /] /system-property=test:add(value="test123")
+{"outcome" => "success"}
+----
+
+What happens is:
+
+* The change is applied to the configuration file.
+* The configuration file is added to a new commit.
+
+
+[[local_git_repository]]
+== Local Git Repository
+
+Starting the server with the option `--git-repository=local` will initiate a Git repository if none exists or use the current Git repository.When initiating the local Git repository a `.gitignore` file will be created and added to the initial commit.
+If a `--git-branch` parameter is added then the repository checkout to this branch and commit to it, otherwise it will default to `master`. Please note that we don't create branch, so it is up to the user to create it beforehand.
+
+[[remote_git_repository]]
+== Remote Git Repository
+
+If a remote Git repository is provided then the server will try to pull from it at boot. If this is the first time we are pulling then local files will be deleted to avoid the pull to fail because of the need to overwrite those existing files.
+The parameter for `git-repository` can be a url or a remote alias provided you have manually added it to the local git configuration.
```

Comment:
"git-repository" should be prefixed with "--". And should be "--git-repo" if the short form is correct.